libfirm-rs/src/nodes_gen.rs

```diff
@@ -0,0 +1,2786 @@
+use libfirm_rs_bindings as bindings;
+use std::collections::HashMap;
+use super::nodes::NodeTrait;
+
+#[derive(Debug)]
+pub enum Node {
```

Comment:
In the current `libfirm_rs` wrappers, each Node type (at least the ones we use) is a seperate struct, not a variant of one big `Node` enum. That makes it possible to define traits for "groups" of nodes, such as the `ValueNode` or `ALUOperand` traits, so that certain functions can only accepts Nodes implementing that trait. 

The advantage is that the compiler complains if you use the wrong node in the wrong place.

I suspect that this is more difficult (impossible) to completely autogenerate, because the information of which function accepts which kind of node is only menthioned in the libfirm docs (if at all)